src/algebra/archimedean.lean

```diff
@@ -78,6 +78,79 @@ begin
   exact abs_sub_lt_iff.2 ⟨by linarith, by linarith⟩
 end
 
+lemma floor_eq_iff {r : α} {z : ℤ} :
+  ⌊r⌋ = z ↔ ↑z ≤ r ∧ r < (z + 1) :=
+by rw [←le_floor, ←int.cast_one, ←int.cast_add, ←floor_lt,
+int.lt_add_one_iff, le_antisymm_iff, and.comm]
+
+/-- The fractional part fract r of r is just r - ⌊r⌋ -/
+def fract (r : α) : α := r - ⌊r⌋
+
+-- Mathematical notation is usually {r}. Let's not even go there.
+-- I have no notation currently.
+
+@[simp] theorem fract_add_floor (r : α) : (⌊r⌋ : α) + fract r = r := by unfold fract;simp
```

Comment:
Missing space after semicolon (maybe on next line, check if your line is more than 100 characters).

Also, if this is a simp lemma, then probably also `floor_add_fract` should be a simp lemma, for completeness.

Reply:
Added `floor_add_fract` and the space. It's just under 100 chars.

*D'oh* how have I managed to screw this up again?